src/ui/SessionExpiryModal/SessionExpiryModal.tsx

```diff
@@ -0,0 +1,77 @@
+import Cookies from 'js-cookie'
+import React, { useEffect, useState } from 'react'
+
+import Button from 'ui/Button'
+import Modal from 'ui/Modal'
+
+const ONE_MINUTE_MILLIS = 60 * 1000
+const TWO_MINUTES_MILLIS = 2 * ONE_MINUTE_MILLIS
+const THIRTY_MINUTES_MILLIS = 30 * ONE_MINUTE_MILLIS
+
+const SessionExpiryModal: React.FC = () => {
+  const [showModal, setShowModal] = useState(false)
+  const sessionExpiryTimeString = Cookies.get('session_expiry')
+
+  const checkSession = (sessionExpiryTime: Date) => {
+    const currentTime = new Date()
+    const timeLeft = sessionExpiryTime.getTime() - currentTime.getTime()
+    if (timeLeft <= TWO_MINUTES_MILLIS) {
+      setShowModal(true)
+    }
+  }
+
+  useEffect(() => {
+    if (!sessionExpiryTimeString) {
+      return
+    }
+
+    const sessionExpiryTime = new Date(sessionExpiryTimeString)
+    const timeLeft = sessionExpiryTime.getTime() - new Date().getTime()
+    const delayBeforeStart =
+      timeLeft > THIRTY_MINUTES_MILLIS ? timeLeft - THIRTY_MINUTES_MILLIS : 0
+
+    const setupSessionIntervalCheck = (sessionExpiryTime: Date) => {
+      checkSession(sessionExpiryTime)
+      return setInterval(
+        () => checkSession(sessionExpiryTime),
+        ONE_MINUTE_MILLIS
+      )
+    }
+
+    let intervalId: NodeJS.Timeout
```

Comment:
This will be running in the browser so it shouldn't use a node related type. Timeout interval ids are just numbers so you should be able to just set it as a type `number`